### ISSUE TITLE
Adding Logger & Its Future Scope

### ISSUE THREAD

**Issue by prabhuomkar** [siesgstarena/Epicentre#1]
Add **winston logger** with following features:
 - _Log File Rotation_ with custom time period via config/env variable (consider backup, maxAge, etc.)
 - _Log File Name_ via config/env variable
 - _Log Format_ (both JSON/TEXT) via config/env variable

 Also, allow stdout to both console and file

**Comment by prabhuomkar** [comment]
cc @Swap76 what should we do for this?

**Comment by Swap76** [comment]
Cool I will start working on it. Just one question where you want the backups to be stored and after how much time ?

**Comment by prabhuomkar** [comment]
@Swap76 maybe assign someone else?
Logs can be stored in Firebase storage just as the other Firebase configuration given in the project.

**Comment by Swap76** [comment]
Sure I will have talk with barath and mithil. Which issue they are working on and assign accordingly

**Comment by ShabarishRamaswamy** [comment]
I am currently looking at how to implement the required features.

**Comment by ShabarishRamaswamy** [comment]
I have a  question regarding this :
What all things do we need to log (all file names, all folders, uploaded status? ).

**Comment by prabhuomkar** [comment]
> What all things do we need to log (all file names, all folders, uploaded status? ).

Every console.log, console.err, logs on startup, connection with services and any webhook push which comes to our services should be logged.
Goal:
With logs we can trace back if there is any issue specific to any service or add some code to tackle during such issues.

**Comment by ShabarishRamaswamy** [comment]
Okay, on it!

**Comment by ShabarishRamaswamy** [comment]
Almost done with the logger part, currently it is logging two levels :
**Error** as well as **Info**.
Please do tell me how do I test the individual API controllers for logging.

**Comment by prabhuomkar** [comment]
@ShabarishRamaswamy test following features:
- logging level be configured via config file
- log file created with a name and log file rotation after X days/hours,etc.
- file should have console/JSON with all details of any event

**Comment by Swap76** [comment]
@prabhuomkar We are facing some issues while working with Typescript and integrating the modules of the logger with rest of the project

Below is the function created to log messages and exported as shown
<img width="429" alt="Screenshot 2020-04-05 at 5 49 13 PM" src="https://user-images.githubusercontent.com/37082605/78533119-0a875680-7806-11ea-9809-66cc6f4ed6b7.png">

In the backup.ts file we have imported the function as below
<img width="270" alt="Screenshot 2020-04-05 at 5 49 25 PM" src="https://user-images.githubusercontent.com/37082605/78533178-2b4fac00-7806-11ea-955b-60f98065a01f.png">

vscode Identifies the function from the import as to shown below
<img width="550" alt="Screenshot 2020-04-05 at 5 49 37 PM" src="https://user-images.githubusercontent.com/37082605/78533268-53d7a600-7806-11ea-9172-dba8eeed0f8e.png">

But still, when we try to the application the below error occurs
<img width="763" alt="Screenshot 2020-04-05 at 5 47 52 PM" src="https://user-images.githubusercontent.com/37082605/78533343-6fdb4780-7806-11ea-850e-be6eca196927.png">

We have tried almost all ways of exporting module.exports, export default etc. But the same problem persists

I have even tried running the application with tsnode rather than normal tsc but still the error persists

**In Short**

The typescript is not understanding out exports and imports

**Comment by prabhuomkar** [comment]
Closing this as working on v2.0
